compiler/rustc_middle/src/query/mod.rs

```diff
@@ -2257,6 +2257,15 @@ rustc_queries! {
     query find_field((def_id, ident): (DefId, rustc_span::symbol::Ident)) -> Option<rustc_target::abi::FieldIdx> {
         desc { |tcx| "find the index of maybe nested field `{ident}` in `{}`", tcx.def_path_str(def_id) }
     }
+
+    /// Construct a type for a trait object corresponding to `trait_ref`. This type will have all
```

Comment:
Why does this need to be a query?

Reply:
Like, this should just live in `compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs`

Reply:
This was a query for two reasons:
1. My previous approach called it from places where depending on `rustc_infer` would make a dependency loop (this isn't the case anymore, so it doesn't hold).
2. We call this once per method in the vtable, so this provides a cache.

Neither of these are strong reasons to keep it a query anymore, so I moved it out like you suggested.